pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -67,28 +66,22 @@ public class GrpcSendingMailbox implements SendingMailbox {
   private final long _deadlineMs;
   private final StatMap<MailboxSendOperator.StatKey> _statMap;
   private final MailboxStatusObserver _statusObserver = new MailboxStatusObserver();
-  private final boolean _splitBlocks;
   private final int _maxByteStringSize;
+  private final Sender _sender;
 
   private StreamObserver<MailboxContent> _contentObserver;
 
   public GrpcSendingMailbox(
       PinotConfiguration config, String id, ChannelManager channelManager, String hostname, int port, long deadlineMs,
-      StatMap<MailboxSendOperator.StatKey> statMap) {
+      StatMap<MailboxSendOperator.StatKey> statMap, boolean splitBlocks, int maxByteStringSize) {
     _id = id;
     _channelManager = channelManager;
     _hostname = hostname;
     _port = port;
     _deadlineMs = deadlineMs;
     _statMap = statMap;
-    _splitBlocks = config.getProperty(
-        CommonConstants.MultiStageQueryRunner.KEY_OF_ENABLE_DATA_BLOCK_PAYLOAD_SPLIT,
-        CommonConstants.MultiStageQueryRunner.DEFAULT_ENABLE_DATA_BLOCK_PAYLOAD_SPLIT);
-    // so far we ensure payload is not bigger than maxBlockSize/2, we can fine tune this later
-    _maxByteStringSize = Math.max(config.getProperty(
-        CommonConstants.MultiStageQueryRunner.KEY_OF_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES,
-        CommonConstants.MultiStageQueryRunner.DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES
-    ) / 2, 1);
+    _maxByteStringSize = maxByteStringSize;
```

Comment:
IICU `_maxByteStringSize` is never used in this class. It can be part of `SplitSender`

Reply:
You're right, moved it to a private member in the sender that requires it.